command/repo.go

```diff
@@ -63,6 +83,162 @@ func repoClone(cmd *cobra.Command, args []string) error {
+	if err != nil {
+		return err
+	}
+
+	apiClient, err := apiClientForContext(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %w", err)
+	}
+
+	var toFork ghrepo.Interface
+	inParent := false // whether or not we're forking the repo we're currently "in"
+	if len(args) == 0 {
+		baseRepo, err := determineBaseRepo(cmd, ctx)
+		if err != nil {
+			return fmt.Errorf("unable to determine base repository: %w", err)
+		}
+		inParent = true
+		toFork = baseRepo
+	} else {
+		repoArg := args[0]
+
+		if isURL(repoArg) {
+			parsedURL, err := url.Parse(repoArg)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+			toFork, err = ghrepo.FromURL(parsedURL)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+		} else {
+			toFork = ghrepo.FromFullName(repoArg)
+			if toFork.RepoName() == "" || toFork.RepoOwner() == "" {
+				return fmt.Errorf("could not parse owner or repo name from %s", repoArg)
+			}
+		}
+	}
+
+	greenCheck := utils.Green("✓")
+	out := colorableOut(cmd)
+	s := utils.Spinner()
+	loading := utils.Gray("Forking ") + utils.Bold(utils.Gray(ghrepo.FullName(toFork))) + utils.Gray("...")
+	s.Suffix = " " + loading
+	s.FinalMSG = utils.Gray(fmt.Sprintf("- %s\n", loading))
+	s.Start()
+
+	authLogin, err := ctx.AuthLogin()
+	if err != nil {
+		s.Stop()
+		return fmt.Errorf("could not determine current username: %w", err)
+	}
+
+	possibleFork := ghrepo.New(authLogin, toFork.RepoName())
+
+	forkedRepo, err := api.ForkRepo(apiClient, toFork)
+	if err != nil {
+		s.Stop()
+		return fmt.Errorf("failed to fork: %w", err)
+	}
+
+	s.Stop()
+	// This is weird. There is not an efficient way to determine via the GitHub API whether or not a
+	// given user has forked a given repo. We noticed, also, that the create fork API endpoint just
+	// returns the fork repo data even if it already exists -- with no change in status code or
+	// anything. We thus check the created time to see if the repo is brand new or not; if it's not,
+	// we assume the fork already existed and report an error.
+	created_ago := Since(forkedRepo.CreatedAt)
+	if created_ago > time.Minute {
+		fmt.Fprintf(out, "%s %s %s\n",
+			utils.Yellow("!"),
+			utils.Bold(ghrepo.FullName(possibleFork)),
+			"already exists")
+	} else {
+		fmt.Fprintf(out, "%s Created fork %s\n", greenCheck, utils.Bold(ghrepo.FullName(forkedRepo)))
+	}
+
+	if (inParent && remotePref == "false") || (!inParent && clonePref == "false") {
+		return nil
+	}
+
+	if inParent {
+		remoteDesired := remotePref == "true"
+		if remotePref == "prompt" {
+			err = Confirm("Would you like to add a remote for the fork?", &remoteDesired)
+			if err != nil {
+				return fmt.Errorf("failed to prompt: %w", err)
+			}
+		}
+		if remoteDesired {
+			_, err := git.AddRemote("fork", forkedRepo.CloneURL, "")
+			if err != nil {
+				return fmt.Errorf("failed to add remote: %w", err)
+			}
+
+			fetchCmd := git.GitCommand("fetch", "fork")
```

Comment:
Just noticed that `AddRemote` already does auto-fetch, so we don't have to do `git fetch` here